Helix api support

zm3y opened the pull request:
This is my try to migrate this extension to Helix api. Maby it had some bugs idk. 

**What work:**
- Authorisation
- Notifications & Notification settings
- Top games (now twitch call it as categories) and view streams by category
- Top streams
- Channels search 
- Filter by language and name

**What removed/changed/problems:**

- Removed Followed games (removed from api)
- Removed Vodcast/rerun check (helix api dont have this info O_o, this is strange)
- Bug Pagination/load more streams (work only on notification settings)
- No channel logos in notification settings & offline channels (need to do request for each logo, but rate limit only 30 in 1 min)
- No info (viewers & channels couter) about top categories (api dont have this info)
- Removed videos from category view (api send emty or very old videos)
- Follow/unfollow dont work (removed from api)

zm3y committed:
Allow disable split notifications

zm3y committed:
Merge branch 'Ndragomirov:master' into master

zm3y committed:
Helix auth point & permissions

zm3y committed:
Helix auth point & permissions for firefox build

zm3y committed:
Disable followed games

zm3y committed:
Changes for tranfer to Helix api

zm3y committed:
Change endpoints & responses processing for Helix api

zm3y committed:
Display data from Helix api

zm3y committed:
Display data from Helix api

zm3y committed:
Display data from Helix api & disable video view

zm3y committed:
Display data in stream view from Helix api

nthnwllms commented:
Amazing work 👏

I built it and it looks to be working, even with the v5 API being down at the moment

![image](https://user-images.githubusercontent.com/17016985/154359866-3abf7547-2ae1-4865-b413-6bdc87235cc6.png)

I'll continue to use this and test. I don't use features like notifications, I only use it to see the number of streams that are live and access them but I'll experiment a bit. Thank you very much for your work :)

Abalieno commented:
Can you fork this and republish?

Because I don't think this is being maintained... Maybe it doesn't need any compilation or similar and can just be installed manually (it seems it needs to be built).

volutus commented:
I cloned the repo to test and these changes are working for me during the current planned Kraken outage. Nice work!

![image](https://user-images.githubusercontent.com/37917121/154365464-332e842e-437e-421a-95f2-eaa2b22f9a54.png)

Yames24 commented:
Good stuff! Once I figured out how to build the extension, it works like it did before.

TheKins commented:
Figured out how to build it and it works pretty darn well! Stellar work!

Probably worth a proper fork and re-release on the various stores if the official version doesn't get an update.

TheKins commented:
Filing a possible bug report: Streams appear to always be sorted by viewer count even when the "Sort streams by" setting is changed:

![image](https://user-images.githubusercontent.com/2977232/154479989-d10994ac-aac0-4a07-819a-0ee02260a023.png)
![image](https://user-images.githubusercontent.com/2977232/154480086-a9272fb1-2dec-47cc-976d-76f7e46879b9.png)

zm3y committed:
Fix sorting streams

zm3y commented:
@TheKins forgot about this, thank you, fixed

TheKins commented:
Good _lord_ that was quick. Fix seems to work great, thanks **so** much for doing all this!

Baatho commented:
Compiled to try out from the repository on your page, and have found a bug? (unsure if it's an API issue or extension issue) - streamers with a non-capital letter at the start of their name are being sorted after the capital letters, instead of with them (sorted by title A-Z ofc)

Old, live extension:
![image](https://user-images.githubusercontent.com/1261925/154573996-8c8bbcfa-a56c-480e-8b26-2e2dbb28f8f3.png)

Your Helix fix:
![image](https://user-images.githubusercontent.com/1261925/154573956-49284e2a-8299-4345-9184-920bb1737084.png)

edit: pulled and merged the OG extension and this pull, same issue, just in case there was something else at play

zm3y committed:
Fix sorting by name

zm3y commented:
@qzfive yep, you're right, it's bug. Fixed.

Baatho commented:
That was fast, awesome! Thank you very much for fixing this extension for the new API, it's (imo) by far the best one of its type

TheKins commented:
Another bug report: Sometimes when a channel goes live, it pops up without a name or viewer count and a blank notification. These details return once the list is refreshed.

![image](https://user-images.githubusercontent.com/2977232/154805286-12695dcc-b8f1-41f7-9e8d-ca0864134224.png)
![image](https://user-images.githubusercontent.com/2977232/154805292-41f4593a-19b1-4727-8e31-42ba1a639b81.png)

zm3y commented:
@TheKins I know about this. But when I try catch it, noticed, It was real data from Helix. They return empty user_name field. I keep investigating, maby my fault.

zm3y committed:
Check for empty capitalized name

Up to 60sec after stream start Helix api return empty capitalized streamer name. So we can use streamer login name instead of nothing. Nobody mind?

noviceg commented:
![image](https://user-images.githubusercontent.com/10168433/155204990-f7ae6e35-e955-4f72-8df7-934395cd56c9.png)
niiiice, zm3y is our savior

ExorcistF1 commented:
I have no idea how to build the extension, but i need one

ghost commented:
how to build it?

kinsi55 commented:
I love how Twitch forces a new API version down the throat of thirdparty devs every year only to remove half the features. Thanks for your work.

zm3y commented:
@slevin0007 @ExorcistF1 Compiled for your convenience, hope you know or find how to install/use it
[twitch-now-1.1.220-github-helix-build-chrome.zip](https://github.com/Ndragomirov/twitch-now/files/8120073/twitch-now-1.1.220-github-helix-build-chrome.zip)
I'm not test well firefox, just run and login.
[twitch-now-1.1.220-github-helix-build-firefox.zip](https://github.com/Ndragomirov/twitch-now/files/8120136/twitch-now-1.1.220-github-helix-build-firefox.zip)

Amiell3 commented:
> @slevin0007 @ExorcistF1 Compiled for chrome based browsers for your convenience, hope you know or find how to install/use it [twitch-now-1.1.220-github-helix-build-chrome.zip](https://github.com/Ndragomirov/twitch-now/files/8120073/twitch-now-1.1.220-github-helix-build-chrome.zip)

and for firefox? :3

ghost commented:
> @slevin0007 @ExorcistF1 Compiled for chrome based browsers for your convenience, hope you know or find how to install/use it [twitch-now-1.1.220-github-helix-build-chrome.zip](https://github.com/Ndragomirov/twitch-now/files/8120073/twitch-now-1.1.220-github-helix-build-chrome.zip)

awesome. works perfect

alvesvalentin commented:
Hi,
Do we have an estimated time when this will be merged?

Thx anyway :)

zm3y commented:
@alvesvalentin no, @Ndragomirov not respond, we can only wait and hope he's alive.

alvesvalentin commented:
Okay thx for your answer :)

placchio commented:
First of all, thanks for the work and the zipped version.
A couple of things while im trying it (chrome version):
-when i click on "Top Stream" or game categories it only scroll streamers until a certain point, it doesnt scroll all the way down like the old version (i dont know if this is intended)
-one time when i re-clicked on "top stream", it did this 
![1](https://user-images.githubusercontent.com/83036039/155230898-2a08a123-c02e-4b9b-9479-630639b9b0f9.png)

noviceg commented:
For me desktop notifications usually comes without streamer name and nickname is missing in the extension too after someone goes live, however the nickname returns after a while in the extension (in a few minutes after someone went live), but the worst part is that i cannot see nickname in desktop notifications :( I saw someone was reporting about this and there was some kind of attempt to fix it, but that fix is not working for me I guess..

![image](https://user-images.githubusercontent.com/10168433/155238050-8892720c-4c38-47f4-87cf-4c96c5fd5c7a.png)

SapphireBear commented:
Can someone make a videoguide? What should be downloaded, installed, done?
and how to apply this fix to make extention work?

Yames24 commented:
> Can someone make a videoguide? What should be downloaded, installed, done? and how to apply this fix to make extention work?

scroll up a bit and download either the chrome or firefox build depending on which browser you use or [go here](https://github.com/Ndragomirov/twitch-now/pull/267#issuecomment-1048198384) 

Extract the zip somewhere you can find easily, once you've loaded the extension into your browser you can't move it without needing to reload it. Make sure you select the folder that contains the extension files, sometimes folders get nested when extracting and loading the top level folder in that case won't work.

[Video about extension loading for chrome and firefox](https://www.youtube.com/watch?v=dhaGRJvJAII)

The extension is self hosted so once you've loaded it, you can remove the old version thats on the firefox/google store.

zm3y commented:
@placchio I described this in PR "Bug Pagination/load more streams (work only on notification settings)" . Maby later I make this work, not now.
@noviceg Try my compiled version if this bug exist Il try to fix it ?again? Can anyone confirm this bug? Need more info. I find only empty field on stream start up to 60sec nothig more.

Roolon commented:
> @slevin0007 @ExorcistF1 Compiled for your convenience, hope you know or find how to install/use it [twitch-now-1.1.220-github-helix-build-chrome.zip](https://github.com/Ndragomirov/twitch-now/files/8120073/twitch-now-1.1.220-github-helix-build-chrome.zip) I'm not test well firefox, just run and login. [twitch-now-1.1.220-github-helix-build-firefox.zip](https://github.com/Ndragomirov/twitch-now/files/8120136/twitch-now-1.1.220-github-helix-build-firefox.zip)

Works perfect! Thank you for your work!

DorianSnowball commented:
Just compiled the Opera version from Source and I'm getting 401 'Unauthorized - Missing scope: user:read:follows' after logging in.

zm3y committed:
Helix auth point & permissions for opera build

zm3y commented:
@DorianSnowball Sorry, I'm don't test opera. Now opera build should work. Builded version for opera:
[twitch-now-1.1.220-github-helix-build-opera.zip](https://github.com/Ndragomirov/twitch-now/files/8124095/twitch-now-1.1.220-github-helix-build-opera.zip)

DorianSnowball commented:
Can confirm that it now works for Opera. Awesome! Thank you @zm3y !

noviceg commented:
@zm3y Your compiled version works perfect, I guess I had a version without "Check for empty capitalized name" fix and that's why I had that problem with no streamer nicknames in notifications/extension. Thanks :)

JasperGroeneweg commented:
Perfect, thank you for your work @zm3y !

Amiell3 commented:
I downloaded the extension for firefox 2 times, I get the message that the zip is corrupt

Octaviorios commented:
Yup, the Zip for firefox is indeed not working for me too

zm3y commented:
@AmidoNewsInside @Octaviorios

https://user-images.githubusercontent.com/8388687/155377222-f107bd42-7a4e-4970-8f25-39b4b2493490.mp4

Octaviorios commented:
> @AmidoNewsInside @Octaviorios
> firefox.mp4

Yes, i got it to work like that, but its not a permament solution, as it will be gone as soon as you restart your browser

Nsbx commented:
I think you should make a fork @zm3y (twitch-now-helix ?)
The repo is dead since 2 years ago, @Ndragomirov doesn't have activity from the same laps of time
This is a bit sad but is better than a dead extension.
However, thank for your PR i'm a bit lost since mine stopped work

kittenchilly commented:
> I think you should make a fork @zm3y (twitch-now-helix ?) The repo is dead since 2 years ago, @Ndragomirov doesn't have activity from the same laps of time This is a bit sad but is better than a dead extension. However, thank for your PR i'm a bit lost since mine stopped work

I second this

zm3y commented:
@Octaviorios this is firefox restrictions. I cant sign extensions. Only signed extensions wold work like others.
@Nsbx I cant do this now. Republish in chrome/firefox/opera stores requre money and support from me in future. I'm just fixed problem in my fav extension and share to all. I cant promise you all guys this support. Sorry.

Octaviorios commented:
Hey no worries. Thanks for the fixes anyway, at least its working for most people

r57zone commented:
Maybe it would be possible to upload this extension under a different name? Maybe "Twitch Now Plus". It looks like the author is no longer involved in this project.

alvesvalentin commented:
> Maybe it would be possible to upload this extension under a different name? Maybe "Twitch Now Plus". It looks like the author is no longer involved in this project. 

I was thinking about this, i would've contacted @zm3y to discuss it and make it possible

seantalon commented:
> @Octaviorios this is firefox restrictions. I cant sign extensions. Only signed extensions wold work like others. @Nsbx I cant do this now. Republish in chrome/firefox/opera stores requre money and support from me in future. I'm just fixed problem in my fav extension and share to all. I cant promise you all guys this support. Sorry.

You could fork it without publishing to stores, that way others could help to continue development!

vaindil commented:
Thanks a ton for this effort! I've generated a signed xpi for Firefox that should work for everyone. I didn't tweak anything from @zm3y's build, just submitted it for signing since I already have an account set up for that. It's not listed publicly on the store either.

Extract the .xpi from this zip file, then drag it onto `about:addons` in Firefox (the normal addons page).

[twitch_now_github_helix_build-1.1.220-an+fx.xpi.zip](https://github.com/Ndragomirov/twitch-now/files/8128008/twitch_now_github_helix_build-1.1.220-an%2Bfx.xpi.zip)

Octaviorios commented:
> Thanks a ton for this effort! I've generated a signed xpi for Firefox that should work for everyone. I didn't tweak anything from @zm3y's build, just submitted it for signing since I already have an account set up for that. It's not listed publicly on the store either.
> 
> Extract the .xpi from this zip file, then drag it onto `about:addons` in Firefox (the normal addons page).
> 
> [twitch_now_github_helix_build-1.1.220-an+fx.xpi.zip](https://github.com/Ndragomirov/twitch-now/files/8128008/twitch_now_github_helix_build-1.1.220-an%2Bfx.xpi.zip)

You are awesome. Thanks! This works! I really hope this extension doesn't ever die, i love it.

bilowik commented:
> Thanks a ton for this effort! I've generated a signed xpi for Firefox that should work for everyone. I didn't tweak anything from @zm3y's build, just submitted it for signing since I already have an account set up for that. It's not listed publicly on the store either.
> 
> Extract the .xpi from this zip file, then drag it onto `about:addons` in Firefox (the normal addons page).
> 
> [twitch_now_github_helix_build-1.1.220-an+fx.xpi.zip](https://github.com/Ndragomirov/twitch-now/files/8128008/twitch_now_github_helix_build-1.1.220-an%2Bfx.xpi.zip)

Works perfectly thanks!
And for anyone that may be a bit skeptical trusting a rando on the internet, I ran a recursive diff and none of the code was changed in that zip, so it's safe!

DanzoGit commented:
@zm3y
You can get stream status from here: https://api.twitch.tv/helix/streams
and from here: https://api.twitch.tv/helix/streams/followed
If the first one (https://api.twitch.tv/helix/streams) returns an empty array
And the second (https://api.twitch.tv/helix/streams/followed) returns "live"
So this is a rerun.
This way you can distinguish rerun from live.

vct4 commented:
Please help.
After installing the Chrome version manually, when I restart the browser, the entire folder of files disappears.

Nsbx commented:
> Please help. After installing the Chrome version manually, when I restart the browser, the entire folder of files disappears.

That totally normal, loading unpacked extension is for debugging purpose only. You should install the extension via xpi version
https://github.com/Ndragomirov/twitch-now/pull/267#issuecomment-1049247939

vct4 commented:
> Please help. After installing the Chrome version manually, when I restart the browser, the entire folder of files disappears.

Was self resolved.
I don't know why, but it seems to be deleted when placed in the \ Google \ Chrome \ User Data \ Default \ Extensions folder and not deleted when placed in another directory.

placchio commented:
Two suggestions:
- I noticed permabanned channels still show up in the list (its impossible to unfollow in your twitch account because permabanned channel dont show up). Its not a huge problem UNLESS there is still the 100 streamers scrolling limit;
- Maybe there is a way to add an option to hide replay/vodcast.

kinsi55 commented:
Might be a good time to also limit the URL's the extension can access instead of using a wildcard match if reuploading under a different name is something thats happening

vaindil commented:
@zm3y Hey, I know you said you just wanted to get this fix in so it continues working but you can't support it long-term. Would you be okay with me forking your code and continuing to maintain it? Don't know how much time I'll be able to spend on it, but I'm willing to get it listed on stores and whatnot and keep it running long-term with some fixes like @kinsi55 mentioned (since I noticed how wide open the permissions are too).

zm3y commented:
Guys you all can use this code how you want. But dont forget about credits to all who work on this. I'm in Ukraine and you all know what happening here. Take care of yourself.

padhie commented:
i would be happy if this hard work will be publish and valued from everyone :) 
my support you get 👍 for this change (the Plugin) and the current situation too

MeekSquid commented:
> > Can someone make a videoguide? What should be downloaded, installed, done? and how to apply this fix to make extention work?
> 
> scroll up a bit and download either the chrome or firefox build depending on which browser you use or [go here](https://github.com/Ndragomirov/twitch-now/pull/267#issuecomment-1048198384)
> 
> Extract the zip somewhere you can find easily, once you've loaded the extension into your browser you can't move it without needing to reload it. Make sure you select the folder that contains the extension files, sometimes folders get nested when extracting and loading the top level folder in that case won't work.
> 
> [Video about extension loading for chrome and firefox](https://www.youtube.com/watch?v=dhaGRJvJAII)
> 
> The extension is self hosted so once you've loaded it, you can remove the old version thats on the firefox/google store.

thanks

CodexHere commented:
Dunno if you're wanting to maintain this or not, but as others have said, the original maintainer has abandoned the project.

You've done the heavy lifting, and then-some, AND the original source is MIT.

I vote publish it, maybe as `Twitch Now (Helix)`...

And if need-be, mass report to get it removed from the store so others don't install a dead/defunct/abandoned extension: 
https://chrome.google.com/webstore/report/nlmbdmpjmlijibeockamioakdpmhjnpk

----

EDIT:

Also, for posterity sake, it might be beneficial to post a set of zips to the release section of your repo for others to find easily

webfischi commented:
> > Thanks a ton for this effort! I've generated a signed xpi for Firefox that should work for everyone. I didn't tweak anything from @zm3y's build, just submitted it for signing since I already have an account set up for that. It's not listed publicly on the store either.
> > Extract the .xpi from this zip file, then drag it onto `about:addons` in Firefox (the normal addons page).
> > [twitch_now_github_helix_build-1.1.220-an+fx.xpi.zip](https://github.com/Ndragomirov/twitch-now/files/8128008/twitch_now_github_helix_build-1.1.220-an%2Bfx.xpi.zip)
> 
> You are awesome. Thanks! This works! I really hope this extension doesn't ever die, i love it.

Yes, it's working again! But I'm missing an important feature (the main reason why I use twitch now) the tab that shows the games I'm following is missing :( Is it possible to add that and also upload the addons.mozilla.org ?

kinsi55 commented:
@webfischi as mentioned in the initial PR, Twitch decided to remove that in the new API version because Twitch likes to do such things

pratik-p commented:
Great job on this thanks @zm3y

seantalon commented:
> @zm3y Hey, I know you said you just wanted to get this fix in so it continues working but you can't support it long-term. Would you be okay with me forking your code and continuing to maintain it? Don't know how much time I'll be able to spend on it, but I'm willing to get it listed on stores and whatnot and keep it running long-term with some fixes like @kinsi55 mentioned (since I noticed how wide open the permissions are too).

Make sure you drop a link here whenever you get around to forking

MeekSquid commented:
> Thanks a ton for this effort! I've generated a signed xpi for Firefox that should work for everyone. I didn't tweak anything from @zm3y's build, just submitted it for signing since I already have an account set up for that. It's not listed publicly on the store either.
> 
> Extract the .xpi from this zip file, then drag it onto `about:addons` in Firefox (the normal addons page).
> 
> [twitch_now_github_helix_build-1.1.220-an+fx.xpi.zip](https://github.com/Ndragomirov/twitch-now/files/8128008/twitch_now_github_helix_build-1.1.220-an%2Bfx.xpi.zip)


our firefox hero!!

afourmeaux commented:
Great work, thank you so much !

JulienProjects commented:
@zm3y thanks!

iamoverit commented:
@zm3y any suggestions why it doesnt work in chrome? 
![image](https://user-images.githubusercontent.com/4696087/156781166-f99e7348-4020-470e-9320-ac9cdf54ee50.png)
![image](https://user-images.githubusercontent.com/4696087/156781190-c3418317-9a5c-46d6-a5e1-5f1827dfdc0a.png)

mateusjs commented:
You guys are amazing. 
Thanks for this

1eft0ver commented:
> @zm3y any suggestions why it doesnt work in chrome? ![image](https://user-images.githubusercontent.com/4696087/156781166-f99e7348-4020-470e-9320-ac9cdf54ee50.png) ![image](https://user-images.githubusercontent.com/4696087/156781190-c3418317-9a5c-46d6-a5e1-5f1827dfdc0a.png)

I ran into the same issue on Edge Chromium.
I got a workaround for this, and I think it's worth a shot.
https://github.com/Ndragomirov/twitch-now/issues/265#issuecomment-1060534465

ghost commented:
> > @zm3y any suggestions why it doesnt work in chrome? 
> 
> I ran into the same issue on Edge Chromium. I got a workaround for this, and I think it's worth a shot. [#265 (comment)](https://github.com/Ndragomirov/twitch-now/issues/265#issuecomment-1060534465)

didn't you read that zm3y lives in ukraine? i don't think he has time for that

1eft0ver commented:
> > > @zm3y any suggestions why it doesnt work in chrome?
> > 
> > 
> > I ran into the same issue on Edge Chromium. I got a workaround for this, and I think it's worth a shot. [#265 (comment)](https://github.com/Ndragomirov/twitch-now/issues/265#issuecomment-1060534465)
> 
> didn't you read that zm3y lives in ukraine? i don't think he has time for that

I read that of course.
What does my workaround have to do with that though?
I was providing what I observed so that people can have some discussions of it.
As I said, what I provided was a workaround, not a correct solution IMO.
I don't want to recklessly commit changes just because it works for me personally, even if I don't actually know the root cause of a certain issue.

I've checked the permissions in the manifest.json, and there's indeed a `tabs` permission in it.
I cannot think of any other causes that make the `tabs.onUpdated` event not work as expected for now.
It's welcome If anyone is willing to discuss or provide more information. :blush:

DoctorDiddler commented:
@zm3y Got a link? I'll drop you a fiver for this fix if you'd like.

Marenthyu commented:
Please note that the current implementation leaks the Client Secret of the Application.
This is a Security Vulnerability and allows anyone to create Tokens on behalf of it and Spam the API to death with the Application being at fault.

It should be looked into removing the Client Secret from the Repository by utilizing the [Implicit Auth Flow](https://dev.twitch.tv/docs/authentication/getting-tokens-oauth#oauth-implicit-code-flow) instead, which does not require the Secret.

kittenchilly commented:
I think it broke. It stopped showing streams, and when I tried to log out and log back in it says {"status":500,"message":"invalid authorization"}

Tinchox17 commented:
did it stop working? cant get it to work. work yesterday.

placchio commented:
same issue here, it just stopped working all of a sudden.

seantalon commented:
> Please note that the current implementation leaks the Client Secret of the Application. This is a Security Vulnerability and allows anyone to create Tokens on behalf of it and Spam the API to death with the Application being at fault.
> 
> It should be looked into removing the Client Secret from the Repository by utilizing the [Implicit Auth Flow](https://dev.twitch.tv/docs/authentication/getting-tokens-oauth#oauth-implicit-code-flow) instead, which does not require the Secret.

guessing someone took your advice :(

jannickj commented:
any chance of someone fixing it? :D

noviceg commented:
we need a hero

Zachruff commented:
fingers crossed for a fix

kevinmon91 commented:
I hope somebody can fix it!

jason-e commented:
> Please note that the current implementation leaks the Client Secret of the Application. This is a Security Vulnerability and allows anyone to create Tokens on behalf of it and Spam the API to death with the Application being at fault.
> 
> It should be looked into removing the Client Secret from the Repository by utilizing the [Implicit Auth Flow](https://dev.twitch.tv/docs/authentication/getting-tokens-oauth#oauth-implicit-code-flow) instead, which does not require the Secret.

The extension has used Implicit Auth Flow [for a long time](https://github.com/Ndragomirov/twitch-now/commit/f4dc9c39ff88db5df0291d09dc3b6e97b4474d7a), it seems the client secret was just never removed.

Dittmen111 commented:
It doesn`t work . I can`t log in `twitch now` and error occurs {"status":500,"message":"invalid authorization"}
Please help us ..

zm3y committed:
tabs.onUpdated no url workaround

Thank to [1eft0ver](https://github.com/1eft0ver)

zm3y committed:
New client_id for chrome build

zm3y committed:
New client_id for firefox build

zm3y committed:
New client_id for opera build

zm3y commented:
Hello guys. I'm alive, but who cares :) 
I think twitch just disable @Ndragomirov client_id. I change it to new and it work well.
@1eft0ver Thank you for you find out. Added to last build.
Chrome build : [twitch-now-1.1.221-github-helix-build-chrome.zip](https://github.com/Ndragomirov/twitch-now/files/8381576/twitch-now-1.1.221-github-helix-build-chrome.zip)
Firefox build : [twitch-now-1.1.221-github-helix-build-firefox.zip](https://github.com/Ndragomirov/twitch-now/files/8381591/twitch-now-1.1.221-github-helix-build-firefox.zip) (Pls, anyone, sign this for other people, thank you!)

noviceg commented:
It works! Thank you @zm3y, stay safe!

Temet79 commented:
Thank you @zm3y !

jannickj commented:
thanks @zm3y glad to hear you're still alive! :grin:

vaindil commented:
Thank you for the update! Here's the signed Firefox build.

[twitch_now_github_helix_build-1.1.221-an+fx.xpi.zip](https://github.com/Ndragomirov/twitch-now/files/8382382/twitch_now_github_helix_build-1.1.221-an%2Bfx.xpi.zip)

zm3y commented:
I think I should say this. We all love this extension. It is what we are use. But we need to accept what happened. This extension is dead. I advise everyone to try [Gumbo: Twitch Companion](https://chrome.google.com/webstore/detail/gumbo-twitch-companion/aalmjfpohaedoddkobnibokclgeefamn/). This is a completely new extension from one of the Twitch-Now contributors @Seldszar . Yes, it might be missing something. But it is being actively [developed](https://github.com/Seldszar/Gumbo). It looks very similar and intuitive. Just try. Thank you all, good luck and take care of yourself.

Morten79 commented:
Thanks alot for taken your time getting this extension working again and very happy to hear you're still alive. I wish you all the best.
I'll take a looks at [Gumbo: Twitch Companion](https://chrome.google.com/webstore/detail/gumbo-twitch-companion/aalmjfpohaedoddkobnibokclgeefamn/) it seems like a really good alternative / replacement.
Take care and good luck 💙💛

ghost commented:
Gumbo is by far the best alternative to twitch now.

nad53 commented:
great job there to keep functional this plugin !

Only issue I have found is that whenever I open up Firefox in morning, I have to click on refresh for the plugin to load up in list all streams that are still online.  If I close firefox and re open it, then the streams still show up on the list.

Does this have a way to be fixed ? If not, could be added up as a possible bug fix ?

NeikoNe commented:
Hi, Todaz, the extension is showing error page:
![image](https://user-images.githubusercontent.com/22804117/162048327-5ee7b606-07a1-4247-a4db-d5938c4c3c26.png)

Any ideas about what could be causing it?

Thx for keeping this extension working!!

DimasDSF commented:
> Hi, Todaz, the extension is showing error page:
> ![image](https://user-images.githubusercontent.com/22804117/162048327-5ee7b606-07a1-4247-a4db-d5938c4c3c26.png)
> 
> Any ideas about what could be causing it?
> 
> Thx for keeping this extension working!! 

Did you delete the folder after loading the extension? You need to keep the folder for it to work

NeikoNe commented:
> 

I have not moved it or removed any files but downloading again and adding it into the browser works. Strange... Thx for the suggestion!

y0rune commented:
> Thank you for the update! Here's the signed Firefox build.
> 
> [twitch_now_github_helix_build-1.1.221-an+fx.xpi.zip](https://github.com/Ndragomirov/twitch-now/files/8382382/twitch_now_github_helix_build-1.1.221-an%2Bfx.xpi.zip)

I would like to install it on the Windows, and I have issue.
![obraz](https://user-images.githubusercontent.com/10187235/163543751-29451e43-27a4-446c-b814-2ccdd525923b.png)

oVegaZz commented:
i was using this extension since 2014(6)  and as much I hate changes I think **is time to move on**  😢

y0rune commented:
I sadly started to use the gumbo.

jannickj commented:
:sob:

DanzoGit commented:
I don't like the Gumbo.. There are no sound notifications..

Dracconis commented:
I am having the same issue as NeikoNe and the issue is that when chrome is closed or has an update, the extension folder gets removed from %APPDATA%\Local\Google\Chrome\User Data\Profile 1\Extensions and therefore renders the extension useless. You have to reinstall the extension each and every time this happens.

Ethan-BB commented:
Chrome now errors out with this
![image](https://user-images.githubusercontent.com/12885129/193395579-c0527c66-eba9-4d6d-b2f6-3f22bc85cbcb.png)
Any chance for the update?

Akira0196 commented:
can't use this today....

DimasDSF commented:
Google Chrome forced ManifestV3 some time ago, this extension will no longer work in chrome unless it is remade to work with v3
https://github.com/Ndragomirov/twitch-now/blob/a5db26746d2793322868b1fc3f6450fa246b7773/chrome/manifest.json#L5

Akira0196 commented:
Would anyone be able to help me create this? I really need this program.

DimasDSF commented:
I actually managed to load my fork in chrome just now, it complained about it not being V3 but still loaded in developer mode.
If I remember correctly `zm3y` fixed the breaking API changes that made this repos version not work, I later forked their repo to add a couple more features and autobuild actions.
Try using this version https://github.com/DimasDSF/twitch-now/releases/tag/latest
- Grab `latest-chrome-auto-release.zip`
- Unzip it somewhere you can store it longterm eg. ~/Documents/TwitchNow
- Open Chrome
- Go to extensions manager
- Enable Developer Mode

<img width="1340" height="131" alt="image" src="https://github.com/user-attachments/assets/c2fdd009-ab31-4c83-b964-0bac0b6b47ff" />

- Click `load unpacked` 

<img width="525" height="107" alt="image" src="https://github.com/user-attachments/assets/8b593800-b54d-443f-a303-cb9d74d00328" />

- Select the `...\build\chrome` folder, it should have `_locales` and `common` directories in it

<img width="257" height="177" alt="image" src="https://github.com/user-attachments/assets/58b30c40-8d0f-4e6d-b6c9-e1f60e6c03ca" />


> [!WARNING]
> Just be aware that it will also break as soon as Google decides to no longer let you load V2 extensions

Akira0196 commented:
<img width="611" height="200" alt="image" src="https://github.com/user-attachments/assets/9f07cbde-3faa-4ca2-b38a-8224c417a7e1" />
Thank you for your help, but Chrome still shows the same error:
The extension uses an unsupported manifest version, so it cannot be installed.
Failed to load the manifest.

TheDjRider commented:
In order to continue work, we need an update to Manifest V3 as Chrome forces it, and old extensions can't be loaded even in Dev Mode.

DimasDSF commented:
Just installed a chrome update and got the same error. Guess this is effectively dead on chrome now.

yungsamd17 commented:
Here is simple **Chrome** extension I made that's similar to **_Twitch Now_** and that supports **Manifest V3**.
## **Sam's Twitch Live:**
https://chromewebstore.google.com/detail/sams-twitch-live/fnaolpkjdickppbebcafdajjndmkgbei

<img width="1280" height="800" alt="unnamed" src="https://github.com/user-attachments/assets/ceaabe13-c96b-4e57-9245-b1e7d5f19496" />

Akira0196 commented:
> 這是我製作的簡單的**Chrome**擴充程序，類似於**_Twitch Now_**並且支援**Manifest V3**。
> 
> ## **Sam 的 Twitch Live：**
> https://chromewebstore.google.com/detail/sams-twitch-live/fnaolpkjdickppbebcafdajjndmkgbei
> 
> <img alt="未命名" width="1280" height="800" src="https://private-user-images.githubusercontent.com/64147848/465765145-ceaabe13-c96b-4e57-9245-b1e7d5f19496.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI0NDUzNjQsIm5iZiI6MTc1MjQ0NTA2NCwicGF0aCI6Ii82NDE0Nzg0OC80NjU3NjUxNDUtY2VhYWJlMTMtYzk2Yi00ZTU3LTkyNDUtYjFlN2Q1ZjE5NDk2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzEzVDIyMTc0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI0ODgyM2M5OTUyNmY2NmRjZGNkNDExYjdiMGZkN2MxMjhiNmJiNjhiMWI2YjBkM2E4NzQ3YmMzY2I3NzFmZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4PZGFQSv7Ln8tyQqXUzO-T96BFKxfXZ9CQOOSSju0fk">

Thanks for sharing this extension — it looks great!
However, if the channel name is not in English, the generated URL also uses non-English characters.
For example, if the channel name is '中文', the link becomes twitch.tv/中文.
Would it be possible to fix this?

yungsamd17 commented:
> > 這是我製作的簡單的**Chrome**擴充程序，類似於**_Twitch Now_**並且支援**Manifest V3**。
> > ## **Sam 的 Twitch Live：**
> > https://chromewebstore.google.com/detail/sams-twitch-live/fnaolpkjdickppbebcafdajjndmkgbei
> > <img alt="未命名" width="1280" height="800" src="https://private-user-images.githubusercontent.com/64147848/465765145-ceaabe13-c96b-4e57-9245-b1e7d5f19496.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI0NDUzNjQsIm5iZiI6MTc1MjQ0NTA2NCwicGF0aCI6Ii82NDE0Nzg0OC80NjU3NjUxNDUtY2VhYWJlMTMtYzk2Yi00ZTU3LTkyNDUtYjFlN2Q1ZjE5NDk2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzEzVDIyMTc0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI0ODgyM2M5OTUyNmY2NmRjZGNkNDExYjdiMGZkN2MxMjhiNmJiNjhiMWI2YjBkM2E4NzQ3YmMzY2I3NzFmZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4PZGFQSv7Ln8tyQqXUzO-T96BFKxfXZ9CQOOSSju0fk">
> 
> Thanks for sharing this extension — it looks great! However, if the channel name is not in English, the generated URL also uses non-English characters. For example, if the channel name is '中文', the link becomes twitch.tv/中文. Would it be possible to fix this?

Yes, it'll look into that. Thanks for the reply.

ledcaveman commented:
The fact that this extension still works after all this time is amazing

SGvagon commented:
From today, Helix build Twitch Now suddenly stop working. Is it only api outage or Twitch made some changes to break this addon? Thank you in advance.

EDIT: Now it working again, strange.
